plugin/pkg/scheduler/factory/factory.go

```diff
@@ -123,6 +133,106 @@ func NewConfigFactory(client *client.Client, schedulerName string) *ConfigFactor
 						c.modeler.ForgetPodByKey(t.Key)
 					}
 				})
+
+				pod := obj.(*api.Pod)
+				if pod.Spec.NodeName != "" {
+					c.EquivalencePodCache.SendInvalidNodeCacheReq(pod.Spec.NodeName)
+				}
+			},
+		},
+	)
+
+	c.NodeLister.Store, c.nodePopulator = framework.NewInformer(
```

Comment:
Are all those changes needed in this PR? I think they aren't.
If I am not missing anything, can you please revert them? I would ideally make this PR as small as possible.